version: "3.8"

networks:
  pmb-network:
    external: false

services:
  pmb-service:
    build:
      context: "./"
      dockerfile: Dockerfile.app
    networks:
      - "pmb-network"
    ports:
      - 8080:8080
    depends_on:
      "db":
        condition: service_healthy
    restart: "on-failure"

  pmb-worker:
    environment:
      - OSU_API_CLIENT_ID=${OSU_API_CLIENT_ID}
      - OSU_API_CLIENT_SECRET=${OSU_API_CLIENT_SECRET}
    build:
      context: "./"
      dockerfile: Dockerfile.worker
    networks:
      - "pmb-network"
    depends_on:
      "pmb-service":
        condition: service_started
    restart: "on-failure"

  db:
    image: "postgres:15.2-alpine"
    ports:
      - 5432:5432
    expose:
      - 5432
    networks:
      - "pmb-network"
    environment:
      POSTGRES_USER: db
      POSTGRES_PASSWORD: db
      POSTGRES_DB: db
      PGUSER: db
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready" ]
      interval: 10s
      timeout: 5s
      retries: 5
    restart: "on-failure"

  adminer:
    image: "adminer"
    networks:
      - "pmb-network"
    restart: "on-failure"
    depends_on:
      - "db"
    ports:
      - "8081:8080"